runtime/src/accounts.rs

```diff
@@ -451,19 +451,42 @@ impl Accounts {
             .collect()
     }
 
+    fn filter_zero_lamport_account(
+        account: AccountSharedData,
+        slot: Slot,
+    ) -> Option<(AccountSharedData, Slot)> {
+        if account.lamports > 0 {
+            Some((account, slot))
+        } else {
+            None
+        }
+    }
+
     /// Slow because lock is held for 1 operation instead of many
     pub fn load_slow(
         &self,
         ancestors: &Ancestors,
         pubkey: &Pubkey,
+        is_root_fixed: bool,
     ) -> Option<(AccountSharedData, Slot)> {
-        let (account, slot) = self.accounts_db.load_slow(ancestors, pubkey)?;
+        let (account, slot) = self.accounts_db.load(ancestors, pubkey, is_root_fixed)?;
+        Self::filter_zero_lamport_account(account, slot)
```

Comment:
I don't like this particularly quirk logic insertion here at wrong layer, imo. But this is different story from this pr. and let's just keep existing behavior.